### ISSUE TITLE
Pop-up folder list does not appear in TB 31

### ISSUE THREAD

**Issue by GoogleCodeExporter** [nostalgy/nostalgy#10]
```
What steps will reproduce the problem?
1. Upgrade TB vrom 24.6 to 31.0
2. Enter G (go to folder)
3. Enter folder name or part of it.

What is the expected output? What do you see instead?

I expect a list of matching folder names, but none appears. If I just press 
Return though, the first matching folder is entered, but I am beyond visual 
control.

What version of the product are you using? On what operating system?

The latest version available on the official Mozilla repo is 0.2.31, which is 
what I use.

Please provide any additional information below.


```

Original issue reported on code.google.com by `alexande...@googlemail.com` on 22 Jul 2014 at 11:37

**Comment by mpkossen** [comment]
I can confirm this happening with Thunderbird 31.5.0 on Windows (8.1 64-bit) with Nostalgy 0.2.33.

**Comment by mpkossen** [comment]
Interestingly enough, the box now appears again after having done nothing at all.

**Comment by zabarabab** [comment]
I did not receive that wonder: the problem persists on all clients (some MacOS, some Linux)

**Comment by wezzynl** [comment]
What seemed to work for me was:

- Make the Thunderbird window less high by dragging the window smaller from the bottom of the window
- Hit "g" and type something
- Quit Thunderbird
- Start Thunderbird
- Now it works (even after a restart and resizing the window again)

**Comment by zabarabab** [comment]
This sounds like woodoo to me, but it worked.

**Comment by eliotmoss** [comment]
On 5/6/2015 3:59 PM, Wesly wrote:
> What seemed to work for me was:
>
>   * Make the Thunderbird window less high by dragging the window smaller from the bottom of the window
>   * Hit "g" and type something
>   * Quit Thunderbird
>   * Start Thunderbird
>   * Now it works (even after a restart and resizing the window again)

The list *appears* for me just fine.

But I seem to have lost the ability to select in it using my mouse.  If I
do that, it seems to pick the folder whose screen position is *behind* the
pop-up list, but the item in the list.  I need to type, or to select by
using the up/down arrow keys for a selection in the list to work.  So something
about fielding mouse clicks seems to be off ...

Eliot Moss

**Comment by chrislong** [comment]
I also saw this problem on Ubuntu 14.04.2 LTS with Thunderbird 31.7.0 and Nostalgy 0.2.33.

But when I tried the suggestion above (https://github.com/alainfrisch/nostalgy/issues/10#issuecomment-99588036), it's now working. Very mysterious, but I'm glad it's working again. Can't stand filing my messages by drag & drop. :)

**Comment by jetztgradnet** [comment]
I have the same problem with TB 38.1.0 and Nostalgy 0.2.33 (even with the dev version) on Mac OSX 10.10.4. The text input is validated (it turns red if a folder does not exist and stays black as long as there should be a match) but no popup folder list appears. The trick mentioned above doesn't help.

**Comment by jgoddards** [comment]
I too have the same problem with TB 38.1.0 on Mac OSX 10.10.4. When I choose a folder from the pop-up window, it does not appear in the in the selection window. The trick mentioned above does not help me either.

**Comment by kruess** [comment]
Same here (Win7, TB 38.2.0, Nostalgy 0.2.33): pop-up with matching folder names is missing.

But, in my case, TB window size seems to matter!

- Observed the issue
- resized the TB window (less high, about half the height of the desktop)
- restarted TB
--> pop-up shows up
- maximised & restart
--> pop-up shows up
- reset to approximately the initial height & restart
--> pop-up still shows up

So, for me and now, it's not reproducible, anymore. hmmm

**Comment by twitwi** [comment]
Same problem here (Linux, TB 38.2.0, Nostalgy 0.2.33).
The workaround worked.

**Comment by tpavlic** [comment]
I have two machines -- a Retina-display iMac running El Capitan and a Retina Display MacBook Pro also running El Capitan -- and the pop-up works fine on one (the iMac) and never shows up on the MacBook Pro. The workaround listed here has not worked. 

TB 38.3.0
Nostalgy 0.2.33
OS X 10.11.1

Any new hints?

**Comment by wedeluxe** [comment]
Linux, 64-bit
Thunderbird 38.3.0
Nostalgy 0.2.33

**Short:**
I can confirm this bug to show up. I can confirm the workaround to work.

**In Detail:**
I did a fresh installation of Nostalgy. Fresh means, it has never been installed in the TB profile before.

When trying to move, copy or go to a folder the pop-up list did not show up. Multiple restarts of TB made no difference at all.

Then I tried the workaround presented in https://github.com/alainfrisch/nostalgy/issues/10#issuecomment-99588036 above in this thread. Here's a step-by-step guide to what I did exactly:

+ Demaximize the TB window.
+ Resize the TB window. Adjust its height and width to leave plenty of margin to all four screen edges.
+ Start to go to a different folder by hitting \<G\>.
+ Enter three letters of a really existing folder.
+ Press \<Alt\>+\<F4\> to quit Thunderbird.
+ Start TB again.

Now, when trying to move, copy or go to a folder the pop-up list showed up.

**Comment by tpavlic** [comment]
Just to emphasize, as has now been demonstrated across multiple comments, the workaround in comment #10 seems to only work on Linux and Windows. Resizing, exiting, and re-opening does not fix the problem on OS X.

**Comment by zasran** [comment]
Same problem, workaround does not work.

Nostalgy 0.2.33
Thunderbird 38.4.0
Ubuntu Linux 15.10

**Comment by tpavlic** [comment]
After a while, on OS X, the problem fixed itself. The resize-window trick did not work, but something got fixed over time. Maybe it has to wait for certain indexing processes to finish inside Thunderbird?

**Comment by bill-mcgonigle** [comment]
Same problem, comment 10 workaround does not work in this case.  This is a new machine, new profile, added one IMAP account, installed Nostalgy from AMO, restarted, and get no folder names in the popup with Nostalgy defaults.

Nostalgy 0.2.33
Thunderbird 38.4.0
CentOS 7.2

**Comment by bill-mcgonigle** [comment]
Error console is saying:

Timestamp: 01/12/2016 10:18:20 AM
Error: TypeError: this.view is undefined
Source File: chrome://global/content/bindings/autocomplete.xml
Line: 927

**Comment by bill-mcgonigle** [comment]
#5 might be related.

**Comment by agittins** [comment]
I've had this happen before on my system when it was Fedora 20 and updated Thunderbird, and again now after updating to Fedora 23 (which included a thunderbird update).

In both cases the workaround of reducing window size, blindly accepting a folder, then closing tb resolved it.

**Comment by hkarl** [comment]
Same issue, workaround NOT working. 

* Ubuntu 15.10
* TB 38.6.0
* Nostalgy 0.2.33

(pity, I got it to work on Windows 10 :-/ )

**Comment by ghidinelli** [comment]
Workaround DID work on OS X for me (Retina Macbook Darwin 15.5.0, Nostalgy 0.2.33, TB 38.5.1). I typed in the name of a folder that didn't exist, "something", and then CMD+Q with the text still red. Next restart the pop up appeared.

**Comment by adamkerz** [comment]
OK! I might have some more info: For me, if the config setting (Tools -> Options -> Advanced -> General tab -> Config Editor) `extensions.nostalgy.recent_folders` is an empty string or just `[]`, I get the same behaviour, no pop up list. If I change that to `[\"test\"]`, and restart Thunderbird (if you edit prefs.js directly, remember that closing Thunderbird will overwrite your changes), then everything works as expected!

Must be something to do with Nostalgy creating a pop up window at start up from that list, and if the list is empty, it fails? Hopefully that gives the developers somewhere to start looking and gives users an easier fix than having to blindly choose a folder.

**Comment by alainfrisch** [comment]
Thanks a lot @adamkerz for the investigation.

The problem is that when the suggestion list is empty (which is the case when the search text is empty and there is no "recent folder"), the code still tried to force opening the completion popup (to work around another bug where the popup would not appear).  Apparently, forcing the popup to open when it is empty puts it in a weird state where it refuses to show anything later.

I've pushed a tentative fix (not forcing the popup to open when it is empty) which fixes the issue (at least for the repro case you described).

A snapshot of the development version is available at:

  http://alain.frisch.fr/info/nostalgy-current.xpi

If someone could confirm independently that this solves the issue for them, I'll submit a new official version.

**Comment by MatzeK79** [comment]
Hi Alain,

I can confirm that the tentative fix works in my case. Win7, TB 45.1.1.
The other mentioned workaround did not work.

Thanks!

**Comment by alainfrisch** [comment]
Thanks @MatzeK79 .  I've uploaded a version including the fix for review on addons.mozilla.org.

**Comment by adamkerz** [comment]
v0.2.34 from addons.mozilla.org fixes this bug for me too! Thanks heaps Alain!

**Comment by kruess** [comment]
Tried to force the bug to show up by manually deleting the recent list in config, but after restarting TB w. Nostalgy v0.2.34, everything still works fine. so, unfortunately, I actually cannot test your fix as I don't face the problem, anymore. Actually, it just bumps up occasionally and disappears after a TB restart.

**Comment by jkirker** [comment]
Has anyone gotten Nostalgy to work with Postbox 5?  My folder lists are returning red so I'm not sure they are being indexed.  I'm lost without Nostalgy!!!  Any help or thoughts would be appreciated.  Thanks in advance.

**Comment by jaroslawhartman** [comment]
> Has anyone gotten Nostalgy to work with Postbox 5? My folder lists are returning red so I'm not sure they are being indexed. I'm lost without Nostalgy!!! Any help or thoughts would be appreciated. Thanks in advance.

I have same observation with Postbox 5 and N 0.2.34. What I noticed is that N 0.2.30 downloaded from https://groups.google.com/forum/#!topic/nostalgy-extension/wTUiC3g0oVQ works for me (but this is really old version).

So there must be a kind of regression but I'm not able to find were...

So I have an immediate fix but I'd love to see Nostalgy working off the shelf with Postbox.

Thank you and best regards,
Jarek

**Comment by caugner** [comment]
According to the comments, this issue has been fixed in Nostalgy 0.2.34.
